pkg/daemon/update.go

```diff
@@ -118,6 +117,9 @@ func (dn *Daemon) deleteStaleData(oldConfig, newConfig *mcfgv1.MachineConfig) {
 		}
 		path = filepath.Join(pathSystemd, u.Name)
 		if _, ok := newUnitSet[path]; !ok {
+			if err := dn.disableUnit(u); err != nil {
+				glog.Warningf("Unable to disable %s: %s", u.Name, err)
```

Comment:
That will change the expectation of the function but I'm OK with making that change. Will add.

Reply:
Oh, I didn't notice this was in a different function, everything was just folded together in the review page.

I think I prefer changing this expectation. I've been working under the assumption that anytime something goes wrong and we are unable to make the correct state change, we throw up our hands. It could be that we retry later, or that we ask to be reprovisioned, or both, but it should be bubbled up the call stack so we can make a decision.

Reply:
It might make sense if we break that into a separate pr though, since presumably there are other spots you didn't touch in this function that also aren't bubbling up their errors.

Reply:
After this PR is merged I'll follow up with another to raise up the errors rather than the current flow which eats them.